doc/devel/gitwash/development_workflow.rst

```diff
@@ -158,6 +158,142 @@ When you are ready to ask for the merge of your code:
+
+When working on a PR, changes may occur in the parent branch (usually master).
+This can lead to conflict with changes in your branch. The conflicts can be
+trivial: for example both the parent branch and your branch add an entry to
+the top of `CHANGELOG`.  Git can not unambiguously tell what to with both
+changes (should one go above the other? if so, which order? should it try to
+merge them?) so it declares the branches can not be merged
+cleanly. Github can only automatically merge PR without conflicts, so you will
+need to manually 'rebase'. This is the process of updating your branch with
+upstream changes, and resolving conflicts.
+
+In git, rebasing is a mild form of re-writing history: it effectively forwards
+all your commits to the updated upstream commit. For a much more detailed
+explanation (with pictures!) see `this nice write up
+<http://git-scm.com/book/en/Git-Branching-Rebasing>`.  The numpy team has also
+`documented how to do this
+<http://docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html#rebasing-on-master>`
+In general, re-writing history, particularly published history, is considered
+bad practice, but in this case it is very useful.
+
+The following example assumes that the remote of _your_ github
+repository is called `github` and the remote of the official
+repository is called `matplotlib`.
+
+The first step is to make sure that your local copy of the upstream repository is
+up-to-date::
+
+     $ git fetch matplotlib
+
+This updates your local copy of the repository, but does not change any files
+in your working copy.  Next, switch to the branch that you want to update::
+
+     $ git checkout backend_plt_refactor
+
+You are now ready to start the rebase of your branch onto the target
+parent branch, in this case `matplotlib/master` ::
+
+     $ git rebase matplotlib/master
+
+and git will then give a bunch of feed back::
+
+     First, rewinding head to replay your work on top of it...
+     Applying: first steps to extract FigureManager* and friends from pyplot
+     Applying: split backend_qt4 into two parts, with and without Gcf
+     ...
+     Applying: pep8 clean up on backend_gtk3.py
+     Using index info to reconstruct a base tree...
+     M       lib/matplotlib/backends/backend_gtk3.py
+     Falling back to patching base and 3-way merge...
+     Auto-merging lib/matplotlib/backends/backend_gtk3.py
+     CONFLICT (content): Merge conflict in lib/matplotlib/backends/backend_gtk3.py
+     Failed to merge in the changes.
+     Patch failed at 0013 pep8 clean up on backend_gtk3.py
+     The copy of the patch that failed is found in:
+        /home/tcaswell/other_source/matplotlib/.git/rebase-apply/patch
+
+     When you have resolved this problem, run "git rebase --continue".
+     If you prefer to skip this patch, run "git rebase --skip" instead.
+     To check out the original branch and stop rebasing, run "git rebase --abort".
+
+A number of commits could be cleanly applied to
```

Comment:
I think this sentence requires an introductory clause, something like: "In the example above, "